src/Microsoft.ML.ImageAnalytics/ImageLoader.cs

```diff
@@ -204,8 +251,107 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
+            {
+                int count = -1;
+                int bytesread = -1;
+                // bufferSize == 1 used to avoid unnecessary buffer in FileStream
+                using (FileStream fs = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1))
+                {
+                    long fileLength = fs.Length;
+                    if (fileLength > int.MaxValue)
+                        throw new IOException($"File {path} too big to open.");
+                    else if (fileLength == 0)
+                    {
+                        Console.WriteLine("File length is zero even though the image is not empty");
+                        byte[] imageBuffer;
+                        // Some file systems (e.g. procfs on Linux) return 0 for length even when there's content.
+                        // Thus we need to assume 0 doesn't mean empty.
+                        imageBuffer = File.ReadAllBytes(path);
+                        count = imageBuffer.Length;
+                        imgData = new VBuffer<byte>(count,imageBuffer);
+                        return count;
+                    }
+
+                    count = (int)fileLength;
+                    var editor = VBufferEditor.Create(ref imgData, count);
+
+#if NETSTANDARD2_0
+                    bytesread = ReadToEnd(fs, editor.Values);
+                    Contracts.Assert(count == bytesread);
+
+#else
+                    fs.Read(editor.Values);
+                    bytesread = editor.Values.Length;
+                    Contracts.Assert(count == bytesread);
+#endif
+                    imgData = editor.Commit();
+                    return count;
+
+                }
+
+            }
+
+            public static int ReadToEnd(System.IO.Stream stream, Span<byte> bufferspan)
+            {
+
+                int chunksize = 4096; // Most optimal size for buffer, friendly to CPU's L1 cache
+                byte[] readBuffer = new byte[chunksize];
+                int totalBytesRead = 0;
+                int bytesRead;
+                unsafe
+                {
+                    fixed (byte* readBufferPtr = readBuffer)
+                    fixed (byte* bufferSpanPtr = &MemoryMarshal.GetReference(bufferspan))
+                    {
+                        while ((bytesRead = stream.Read(readBuffer, 0, readBuffer.Length)) > 0)
+                        {
+                            Buffer.MemoryCopy(readBufferPtr, bufferSpanPtr + totalBytesRead, bufferspan.Length - totalBytesRead, bytesRead);
+                            totalBytesRead += bytesRead;
+                        }
+                    }
+                }
+                return totalBytesRead;
+
+            }
+
+            public DataViewType GetDataViewType()
+            {
+                if (_type)
+                    return new VectorDataViewType(NumberDataViewType.Byte);
+                else
+                    return  new ImageDataViewType();
+            }
```

Comment:
public DataViewType GetDataViewType() => _type ? new VectorDataViewType(NumberDataViewType.Byte) : new ImageDataViewType();

Reply:
I actually tried that but it is throwing an error. 
"Type of conditional expression cannot be determined because there is no implicit conversion between Microsoft,ML.Data.VectorDataViewType and Microsoft.ML.Transform.Image.ImageDataViewType"

Reply:
Strange